File: BlogWebApp/Blog.Core/Configuration/HostingConfig.cs
// <copyright file="HostingConfig.cs" company="Blog">
// Copyright (c) Blog. All rights reserved.
// </copyright>

namespace Blog.Core.Configuration
{
    /// <summary>
    /// Hosting configuration.
    /// </summary>
    public class HostingConfig
    {
        /// <summary>
        /// Gets or sets forwardedHttpHeader.
        /// </summary>
        public string ForwardedHttpHeader { get; set; }

        /// <summary>
        /// Gets or sets a value indicating whether useHttpClusterHttps.
        /// </summary>
        public bool UseHttpClusterHttps { get; set; }

        /// <summary>
        /// Gets or sets a value indicating whether useHttpXForwardedProto.
        /// </summary>
        public bool UseHttpXForwardedProto { get; set; }
    }
}

File: Shared/Blog.FSharp.Core/Configuration/HostingConfig.fs
namespace Blog.Core.Configuration

/// Hosting configuration.
type HostingConfig = {
    /// Forwarded HTTP header
    ForwardedHttpHeader: string option

    /// Use HTTP cluster HTTPS
    UseHttpClusterHttps: bool

    /// Use HTTP X-Forwarded-Proto
    UseHttpXForwardedProto: bool
}


